### ISSUE TITLE
Connection Timeout

### ISSUE THREAD

**Issue by internpoon** [axios/axios#4835]
#### Is your feature request related to a problem? Please describe.
Yes. Currently I am using Axios **With Proxy** to perform some API request. However, some proxy might be slow or when the response size is too large, Axios will not timeout the request but it will keep waiting and waiting. The timeout option provided by Axios is response timeout, it would be great to have another option like Connection Timeout

#### Describe the solution you'd like
A "Connection Timeout & Retry" would be great.

#### Describe alternatives you've considered
Tried other HTTP client but it wasn't as good as axios in terms of speed. However, the only drawback is the connection timeout :( 

#### Additional context
None.

**Comment by SuperKirik** [comment]
@internpoon maybe this will help...
https://github.com/axios/axios/issues/1739#issuecomment-769148019

**Comment by yunusyuksel** [comment]
> @internpoon maybe this will help... [#1739 (comment)](https://github.com/axios/axios/issues/1739#issuecomment-769148019)

The solution in that comment depends on the node library. Is there any solution for react-native, since react-native does not have any node dependency, that solution does not work

**Comment by leite08** [comment]
I agree we need a connection timeout to axios. This provides greater control on connections.

Our use case is we have a Lambda making an HTTP request and we need to have diff timeouts for connection and response.

I tried the suggestion on https://github.com/axios/axios/issues/1739#issuecomment-769148019 but couldn't make it work (Axios `1.4.0` on Node `18.14.2`). But even with that suggestion, that's work around IMO, since one would expect that sort of control on a HTTP library.

**Comment by jasonsaayman** [comment]
Thanks for your bug report, we will be closing this issue and ask that you please test and try re-create the issue with the latest version of Axios. If the bug is still present please open a issue. If possible and you are willing open a pull request with a proposed fix. Please see the motivation for doing this:

https://github.com/axios/axios/issues/6983
